torch/csrc/jit/tensorexpr/bounds_inference.cpp

```diff
@@ -286,15 +286,17 @@ bool hasPartialOverlap(
     if (bIt == bBounds.end()) {
       continue;
     }
-
-    auto aIndexBounds = getIndexBounds(aBound.second);
-    auto bIndexBounds = getIndexBounds(bIt->second);
-    for (const auto& aIndexBound : aIndexBounds) {
-      for (const auto& bIndexBound : bIndexBounds) {
-        auto overlap = overlaps(aIndexBound, bIndexBound);
-        // If the returned OverlapKind is "Contains", that means `bound1` is
-        // a super set of `bound2`, so that is also a PartialOverlap.
-        if (overlap == Contains || overlap == PartialOverlap) {
+    auto aTABIs = aBound.second;
+    auto bTABIs = bIt->second;
+    for (const auto& aTABI : aTABIs) {
+      for (const auto& bTABI : bTABIs) {
+        if (aTABI.kind == kLoad && bTABI.kind == kLoad) {
+          continue;
+        }
+        auto aIndexBounds = getIndexBounds(aTABI);
+        auto bIndexBounds = getIndexBounds(bTABI);
```

Comment:
Feels pretty suboptimal to do N^2 getIndexBounds here esp since this is doing a bunch of allocations.

Reply:
Good point. Cached them at the start in a map and reused here.